File: ToDoList/app/src/main/java/com/davdo/todolist/anrdoid/MainActivity.java
package com.davdo.todolist.anrdoid;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;

import android.os.Bundle;

import com.davdo.todolist.R;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        FragmentManager fragmentManager = getSupportFragmentManager();
        Fragment fragment = fragmentManager.findFragmentById(R.id.fragment_note_container);

        if(fragment == null) {

//            fragment = new NoteFragment();
            fragment = new NoteListFragment();

//            Bundle testBundle = new Bundle();
//            testBundle.putParcelable(NoteFragment.NOTE_INDEX, note);
//            fragment.setArguments(testBundle);

            fragmentManager.beginTransaction().add(R.id.fragment_note_container, fragment).commit();
        }
    }
}

File: BroadcastReciever/app/src/main/java/com/davdo/broadcastreciever/MainActivity.kt
package com.davdo.broadcastreciever

import android.content.BroadcastReceiver
import android.content.IntentFilter
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle

class MainActivity : AppCompatActivity() {


    private lateinit var reciever : BroadcastReceiver

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        configureReceiver()
    }

    private fun configureReceiver() {

        val filter = IntentFilter()

        filter.addAction("com.davdo.SendBroadcast")
        reciever = MyReciever()
        registerReceiver(reciever, filter)
    }

    override fun onDestroy() {
        super.onDestroy()
        unregisterReceiver(reciever)
    }
}
